### PR TITLE
Form styles

### PR DESCRIPTION
**Pull request by jdmcleod**
[Trello](https://trello.com/c/620gnCPd/324-basic-animation-integration)

Currently this is branched off `add-panels-and-modals` as we were experimenting with adding animations to modals.

This is only a draft pull request right now and the changes have not yet been moved into the css generator and committed there. Right now, the experimental changes are only in the temporary example_rails6 directory.

I would also wait until `color-theory` is merged so we can take advantage of the new gray scale swatch in our updated form styles.

## What changed

We experimented with bringing in [animate.css](https://animate.style/) and using animations in our form. We added a flash animation to the error icon for form inputs and also added a fade-in to the main form error style in our demo-form. I don't imagine that will get merged though.

## Form style updates

- Update inputs and selects and add hover states
- Add checkbox styles hover and disabled states
- Add radio styles 
- Update error state for form inputs
- Updated some button hover states

Screen Shot 2021-02-16 at 5 08 33 PM

### CODE REVIEW

File: lib/generators/rolemodel/css/base/templates/app/javascript/stylesheets/components/checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
+.checkbox {
```
Comment (jdmcleod) [inline lib/generators/rolemodel/css/base/templates/app/javascript/stylesheets/components/checkbox.scss:1]: Jake and I couldn't decide if this should be split into its own `.scss` file or if it should be included in `form.scss` with the class name `.form__checkbox`. The same goes for `radio.scss`

File: example_rails6/app/javascript/stylesheets/variables.scss [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  --color-warning: #856404;
+  --color-warning-light: #ffeeba;
+  --color-warning-lighter: #fff3cd;
+
+  // Contrast
+  --color-contrast-lower: #F9F9F9;
+  --color-contrast-low: #ebebeb;
+  --color-contrast-medium: #E0E0E0;
+  --color-contrast-mediumer: #c4c4c4;
+  --color-contrast-mediumerer: #aaaaaa;
```
Comment (jdmcleod) [inline example_rails6/app/javascript/stylesheets/variables.scss:41]: Replace with `color-theory` work

File: example_rails6/app/javascript/stylesheets/components/form.scss [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+// TODO: Remove this?
+[type="checkbox"] {
+  cursor: pointer;
+}
+
+// DEMO CLASSES
+
+.form--third {
```
Comment (jdmcleod) [inline example_rails6/app/javascript/stylesheets/components/form.scss:18]: remove demo code

**Comment by scriswell** [comment]
@jdmcleod This is an old PR, are we still evaluating this as a feature update? If so we may need to talk about what to merge into all of the work we've done during the latest CCC days and make sure we have a consistent base theme for everything.

**Comment by jdmcleod** [comment]
@scriswell I don't think this is relevant any more. What was wanted from this PR got pulled out a while ago  (checkboxes, radio buttons) and I will close it now.

**Pull request closed without merging**